xl-v1-createfile.go

```diff
@@ -51,31 +51,60 @@ func closeAndRemoveWriters(writers ...io.WriteCloser) {
 // WriteErasure reads predefined blocks, encodes them and writes to
 // configured storage disks.
 func (xl XL) writeErasure(volume, path string, reader *io.PipeReader) {
-	var writers = make([]io.WriteCloser, len(xl.storageDisks))
-	var sha512Writers = make([]hash.Hash, len(xl.storageDisks))
-	var metadataWriters = make([]io.WriteCloser, len(xl.storageDisks))
-
```

Comment:
The order is extremely important in erasure coding, it is better to keep the index and order rather than append over writers it is not safer and can lead to bugs.  Erasure coding problem is that we cannot randomly write blocks, both read and write have tight order. 

While using append() is easier it is really hard to keep order and introduces bugs. At any given time inside CreateFile or ReadFile we cannot exceed the writers or readers respectively other we might read garbage data.

The readers and writers should have an array length exactly similar to `xl.storageDisks` - or `quorumDisks`.